community/graphdb-api/src/main/java/org/neo4j/graphdb/TransactionTerminatedException.java

```diff
@@ -19,26 +19,31 @@
  */
 package org.neo4j.graphdb;
 
-import static java.util.Objects.requireNonNull;
+import org.neo4j.kernel.api.exceptions.Status;
 
 /**
  * Signals that the transaction within which the failed operations ran
  * has been terminated with {@link Transaction#terminate()}.
  */
-public class TransactionTerminatedException extends TransactionFailureException
+public class TransactionTerminatedException extends TransactionFailureException implements Status.HasStatus
 {
-    public TransactionTerminatedException()
+    private final Status status;
+
+    public TransactionTerminatedException( Status status )
+    {
+        this( status, "" );
+    }
+
+    protected TransactionTerminatedException( Status status, String additionalInfo )
     {
-        this( "" );
+        super( "The transaction has been terminated.Simply retry your operation in a new transaction, " +
+               "and you should see a successful result." + status.code().description() + " " + additionalInfo );
+        this.status = status;
```

Comment:
You are missing a space in the message, before `Simply`. Also, I don't think the word "simply" adds very much.